ctf_framework/views/analytics.py

```diff
@@ -0,0 +1,44 @@
+from django.contrib.auth.decorators import login_required
+from django.contrib import messages
+from .base_view import *
+from django.urls import reverse
+from django.http import HttpResponseForbidden, HttpResponseNotAllowed
+from ..models import Solve
+from django.utils import timezone
+from datetime import timedelta
+
+@login_required()
+def index(request):
+    """Get some solves metrics."""
+
+    solves = Solve.objects.prefetch_related('challenge', 'user')
+
+    solves_chart_data = {}
+
+    for category in ['Misc', 'Web', 'Pwn', 'Forensics', 'Reverse Engineering', 'Crypto']:
```

Comment:
Suggested refactoring to improve on the following points : 
- Category names shouldn't be hardcoded 
- Variable names `delta_a` and `delta_b` aren't very descriptive
- Personally not a fan of the `__gte` and `__lt` syntax

```
chart_data = {}
now = timezone.now()

for category in Category.objects.all():
  chart_data[category] = []

  for offset in range(7):
    end = now - timedelta(days=offset)
    start = now - timedelta(days=offset + 1)
    solve_count = len([solve for solve in solves if solve.is_between_dates(start, end) and solve.category.name == category])
    chart_data[category].append(solve_count)
 ```

The `is_between_dates` would be a function belonging to the `User` model.

Reply:
done